pronto/api/signatures/go.py

```diff
@@ -95,3 +102,72 @@ def get_go_terms(accessions):
 def _sort_term(term):
     max_prots = max(s["proteins"] for s in term["signatures"].values())
     return -max_prots, term["id"]
+
+def chunks(iterable: Iterable, n):
+    """Yield chunks of size n from iterable."""
+
+    n = max(1, n)
+    return zip(*[iter(iterable)] * n)
+
+
+def get_go2panther(subfams: list, terms: dict, pg_cur):
+    
+    con = utils.connect_oracle()
+    cur = con.cursor()
+    
+    go_terms = set()
+
+    for subfam_set in chunks(subfams, 1000):
+        cur.execute(
+        """
+            SELECT DISTINCT METHOD_AC, GO_ID
+            FROM INTERPRO.PANTHER2GO
+            WHERE METHOD_AC IN ('{}')
+            """.format("','".join(map(str, subfam_set)))
+        )
+
+        for row in cur:
+            term_id = row[1]
+            pth_fam = row[0].split(':')[0]
+
+            try:
+                term = terms[term_id]
+            except KeyError:
+                term = terms[term_id] = {"id":term_id, "signatures":{}}
+                go_terms.add(term_id)
+
+            try:
+                s = term["signatures"][pth_fam]
+            except KeyError:
+                s = term["signatures"][pth_fam] = [set(), set(), set()]
+
+            s[2].add(row[0])
+
+    cur.close()
+    con.close()
+
+    for terms_chunk in chunks(list(go_terms), 100):
+        go_details = get_go_details(terms_chunk, pg_cur)
+        for go, details in go_details.items():
+            terms[go].update(details)
+    return terms
```

Comment:
Using `range()` and modified version of `get_go_details()`:

```suggestion
    go_terms = list(go_terms)
    for i in range(0, len(go_terms), 100):
        subset = go_terms[i:i+100]

        for term in get_go_details(subset, pg_cur):
            terms[term["id"]].update(term)

    return terms
```